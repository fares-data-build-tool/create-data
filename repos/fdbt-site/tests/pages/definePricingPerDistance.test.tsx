import * as React from 'react';
import { shallow } from 'enzyme';
import DefinePricingPerDistance from '../../src/pages/definePricingPerDistance';

describe('pages', () => {
    describe('DefinePricingPerDistance', () => {
        it('should render correctly', () => {
            const tree = shallow(
                <DefinePricingPerDistance
                    errors={[]}
                    csrfToken=""
                    pricingPerDistanceData={{
                        maximumPrice: '4',
                        minimumPrice: '3',
                        capPricing: [
                            {
                                distanceFrom: '0',
                                distanceTo: '2',
                                pricePerKm: '5',
                            },
                            {
                                distanceFrom: '2',
                                distanceTo: 'Max',
                                pricePerKm: '5',
                            },
                        ],
                        productName: 'Product',
                    }}
                    numberOfEntitesByDistanceInitial={1}
                />,
            );
            expect(tree).toMatchSnapshot();
        });

        it('should render error messaging when errors are passed', () => {
            const tree = shallow(
                <DefinePricingPerDistance
                    errors={[
                        {
                            id: `distance-to-0`,
                            errorMessage: 'Enter a value for the distance',
                        },
                    ]}
                    csrfToken=""
                    pricingPerDistanceData={{
                        maximumPrice: '4',
                        minimumPrice: '3',
                        capPricing: [
                            {
                                distanceFrom: '0',
                                distanceTo: '',
                                pricePerKm: '6',
                            },
                            {
                                distanceFrom: '3',
                                distanceTo: 'Max',
                                pricePerKm: '5',
                            },
                        ],
                        productName: 'Product',
                    }}
                    numberOfEntitesByDistanceInitial={2}
                />,
            );
            expect(tree).toMatchSnapshot();
        });
    });
});
