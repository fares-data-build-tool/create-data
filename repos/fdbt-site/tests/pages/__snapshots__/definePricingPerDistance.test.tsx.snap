// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages DefinePricingPerDistance should render correctly 1`] = `
<FullColumnLayout
  description="Pricing Per Distance entry page of the Create Fares Data Service"
  errors={Array []}
  title="Pricing Per Distance - Create Fares Data Service"
>
  <CsrfForm
    action="/api/pricingPerDistance"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={Array []}
    />
    <h1
      className="govuk-heading-l"
      id="cap-capPricing-per-distance-heading"
    >
      Enter your distance details
    </h1>
    <div
      className="govuk-!-margin-bottom-3 govuk-!-margin-left-1"
    >
      <label
        className="govuk-label"
        htmlFor="capped-product-name"
      >
        Product name
      </label>
      <span
        className="govuk-hint"
      >
        Enter the name of your product
      </span>
      <div
        className="govuk-input__wrapper"
      >
        <FormElementWrapper
          errorClass="govuk-input--error"
          errorId="capped-product-name"
          errors={Array []}
          hideText={true}
        >
          <input
            className="govuk-input govuk-input--width-30"
            defaultValue="Product"
            id="capped-product-name"
            name="productName"
            type="text"
          />
        </FormElementWrapper>
      </div>
    </div>
    <div
      className="flex-container"
    >
      <div
        className="govuk-!-margin-left-1 govuk-!-margin-right-2"
      >
        <FormGroupWrapper
          errorIds={
            Array [
              "minimum-price",
            ]
          }
          errors={Array []}
          hideErrorBar={true}
        >
          <label
            className="govuk-label"
            htmlFor="minimum-price"
          >
            Minimum price (£)
          </label>
          <span
            className="govuk-hint"
            id="minimum-price-hint"
          >
            e.g. £2.99
          </span>
          <div
            className="govuk-input__wrapper"
          >
            <div
              aria-hidden="true"
              className="govuk-input__prefix"
            >
              £
            </div>
            <FormElementWrapper
              errorClass="govuk-input--error"
              errorId="minimum-price"
              errors={Array []}
              hideText={true}
            >
              <input
                className="govuk-input govuk-input--width-3"
                id="minimum-price"
                name="minimumPrice"
                onChange={[Function]}
                type="text"
                value="3"
              />
            </FormElementWrapper>
          </div>
        </FormGroupWrapper>
      </div>
      <div
        className="govuk-!-margin-left-2 govuk-!-margin-right-2"
      >
        <FormGroupWrapper
          errorIds={
            Array [
              "maximum-price",
            ]
          }
          errors={Array []}
          hideErrorBar={true}
        >
          <label
            className="govuk-label"
            htmlFor="maximum-price"
          >
            Maximum price (£)
          </label>
          <span
            className="govuk-hint"
            id="maximum-price-hint"
          >
            e.g. £2.99
          </span>
          <div
            className="govuk-input__wrapper"
          >
            <div
              aria-hidden="true"
              className="govuk-input__prefix"
            >
              £
            </div>
            <FormElementWrapper
              errorClass="govuk-input--error"
              errorId="maximum-price"
              errors={Array []}
              hideText={true}
            >
              <input
                className="govuk-input govuk-input--width-3"
                id="maximum-price"
                name="maximumPrice"
                onChange={[Function]}
                type="text"
                value="4"
              />
            </FormElementWrapper>
          </div>
        </FormGroupWrapper>
      </div>
    </div>
    <div
      className="govuk-grid-row"
    >
      <DistanceRow
        errors={Array []}
        numberOfCapToDisplay={1}
        pricingPerDistance={
          Object {
            "capPricing": Array [
              Object {
                "distanceFrom": "0",
                "distanceTo": "2",
                "pricePerKm": "5",
              },
              Object {
                "distanceFrom": "2",
                "distanceTo": "Max",
                "pricePerKm": "5",
              },
            ],
            "maximumPrice": "4",
            "minimumPrice": "3",
            "productName": "Product",
          }
        }
        setPricingPerDistance={[Function]}
      />
      <div
        className="flex-container"
      >
        <button
          className="govuk-button govuk-button--secondary govuk-!-margin-left-3 govuk-!-margin-bottom-3 time-restrictions-button-placement"
          id="add-another-button"
          onClick={[Function]}
          type="button"
        >
          Add another row
        </button>
      </div>
      <input
        className="govuk-button govuk-!-margin-left-3"
        id="continue-button"
        type="submit"
        value="Continue"
      />
    </div>
  </CsrfForm>
</FullColumnLayout>
`;

exports[`pages DefinePricingPerDistance should render error messaging when errors are passed 1`] = `
<FullColumnLayout
  description="Pricing Per Distance entry page of the Create Fares Data Service"
  errors={
    Array [
      Object {
        "errorMessage": "Enter a value for the distance",
        "id": "distance-to-0",
      },
    ]
  }
  title="Pricing Per Distance - Create Fares Data Service"
>
  <CsrfForm
    action="/api/pricingPerDistance"
    csrfToken=""
    method="post"
  >
    <ErrorSummary
      errors={
        Array [
          Object {
            "errorMessage": "Enter a value for the distance",
            "id": "distance-to-0",
          },
        ]
      }
    />
    <h1
      className="govuk-heading-l"
      id="cap-capPricing-per-distance-heading"
    >
      Enter your distance details
    </h1>
    <div
      className="govuk-!-margin-bottom-3 govuk-!-margin-left-1"
    >
      <label
        className="govuk-label"
        htmlFor="capped-product-name"
      >
        Product name
      </label>
      <span
        className="govuk-hint"
      >
        Enter the name of your product
      </span>
      <div
        className="govuk-input__wrapper"
      >
        <FormElementWrapper
          errorClass="govuk-input--error"
          errorId="capped-product-name"
          errors={
            Array [
              Object {
                "errorMessage": "Enter a value for the distance",
                "id": "distance-to-0",
              },
            ]
          }
          hideText={true}
        >
          <input
            className="govuk-input govuk-input--width-30"
            defaultValue="Product"
            id="capped-product-name"
            name="productName"
            type="text"
          />
        </FormElementWrapper>
      </div>
    </div>
    <div
      className="flex-container"
    >
      <div
        className="govuk-!-margin-left-1 govuk-!-margin-right-2"
      >
        <FormGroupWrapper
          errorIds={
            Array [
              "minimum-price",
            ]
          }
          errors={
            Array [
              Object {
                "errorMessage": "Enter a value for the distance",
                "id": "distance-to-0",
              },
            ]
          }
          hideErrorBar={true}
        >
          <label
            className="govuk-label"
            htmlFor="minimum-price"
          >
            Minimum price (£)
          </label>
          <span
            className="govuk-hint"
            id="minimum-price-hint"
          >
            e.g. £2.99
          </span>
          <div
            className="govuk-input__wrapper"
          >
            <div
              aria-hidden="true"
              className="govuk-input__prefix"
            >
              £
            </div>
            <FormElementWrapper
              errorClass="govuk-input--error"
              errorId="minimum-price"
              errors={
                Array [
                  Object {
                    "errorMessage": "Enter a value for the distance",
                    "id": "distance-to-0",
                  },
                ]
              }
              hideText={true}
            >
              <input
                className="govuk-input govuk-input--width-3"
                id="minimum-price"
                name="minimumPrice"
                onChange={[Function]}
                type="text"
                value="3"
              />
            </FormElementWrapper>
          </div>
        </FormGroupWrapper>
      </div>
      <div
        className="govuk-!-margin-left-2 govuk-!-margin-right-2"
      >
        <FormGroupWrapper
          errorIds={
            Array [
              "maximum-price",
            ]
          }
          errors={
            Array [
              Object {
                "errorMessage": "Enter a value for the distance",
                "id": "distance-to-0",
              },
            ]
          }
          hideErrorBar={true}
        >
          <label
            className="govuk-label"
            htmlFor="maximum-price"
          >
            Maximum price (£)
          </label>
          <span
            className="govuk-hint"
            id="maximum-price-hint"
          >
            e.g. £2.99
          </span>
          <div
            className="govuk-input__wrapper"
          >
            <div
              aria-hidden="true"
              className="govuk-input__prefix"
            >
              £
            </div>
            <FormElementWrapper
              errorClass="govuk-input--error"
              errorId="maximum-price"
              errors={
                Array [
                  Object {
                    "errorMessage": "Enter a value for the distance",
                    "id": "distance-to-0",
                  },
                ]
              }
              hideText={true}
            >
              <input
                className="govuk-input govuk-input--width-3"
                id="maximum-price"
                name="maximumPrice"
                onChange={[Function]}
                type="text"
                value="4"
              />
            </FormElementWrapper>
          </div>
        </FormGroupWrapper>
      </div>
    </div>
    <div
      className="govuk-grid-row"
    >
      <DistanceRow
        errors={
          Array [
            Object {
              "errorMessage": "Enter a value for the distance",
              "id": "distance-to-0",
            },
          ]
        }
        numberOfCapToDisplay={2}
        pricingPerDistance={
          Object {
            "capPricing": Array [
              Object {
                "distanceFrom": "0",
                "distanceTo": "",
                "pricePerKm": "6",
              },
              Object {
                "distanceFrom": "3",
                "distanceTo": "Max",
                "pricePerKm": "5",
              },
            ],
            "maximumPrice": "4",
            "minimumPrice": "3",
            "productName": "Product",
          }
        }
        setPricingPerDistance={[Function]}
      />
      <div
        className="flex-container"
      >
        <button
          className="govuk-button govuk-button--secondary govuk-!-margin-left-3 govuk-!-margin-bottom-3 time-restrictions-button-placement"
          id="add-another-button"
          onClick={[Function]}
          type="button"
        >
          Add another row
        </button>
        <button
          className="govuk-button govuk-button--secondary govuk-!-margin-left-3 govuk-!-margin-bottom-3"
          id="remove-button"
          onClick={[Function]}
          type="button"
        >
          Remove last row
        </button>
      </div>
      <input
        className="govuk-button govuk-!-margin-left-3"
        id="continue-button"
        type="submit"
        value="Continue"
      />
    </div>
  </CsrfForm>
</FullColumnLayout>
`;
